aioxmpp/service.py

```diff
@@ -263,14 +258,73 @@ def __get__(self, instance, owner):
         return obj
 
 
+class DependencyGraphNode:
+    def __init__(self):
+        self.class_ = None
+
+    def __repr__(self):
+        return "DependencyGraphNode({!r})".format(self.class_)
+
+
+class DependencyGraph:
+
+    def __init__(self, edges=None, nodes=None):
+        if edges is None:
+            edges = []
+        if nodes is None:
+            nodes = []
+        self._edges = set(edges)
+        self._nodes = set(nodes)
+
+    def copy(self):
+        return DependencyGraph(self._edges, self._nodes)
```

Comment:
Re-implement ``__copy__`` and use ``copy.copy`` instead. Standard mechanisms and such. Although I’m not sure if this doesn’t rather qualify for a ``__deepcopy__`` because it also copies the sets.